extensions/commands/recipe/cmd_bump_deps.py

```diff
@@ -62,7 +62,12 @@ def latest_ref(name: str) -> Optional[str]:
                         ConanOutput().warning(f"Unable to bump non constant dependency in {recipe_file}:{arg.lineno}")
                         continue
                     oldref = arg.value
-                    name = oldref.split("/")[0]
+                    parts = oldref.split("/")
+                    version = parts[1]
+                    if version.startswith("[") or version.endswith("]"):
```

Comment:
I guess it would be a good moment to add support for https://github.com/conan-io/conan/pull/13712